### PR TITLE
Enforce clang 7.0 and gcc 8.2

### PR DESCRIPTION
**Pull request by mrzzzrm**
Hyrise's minimum compiler versions were not properly enforced during configuration.

Dropped minimum clang version to 7.0, since that's still available in Ubuntu 18.04 and seems to build fine.

Raised minimum cmake version to 3.9 - why keep a "(deprecated)" version in DEPENEDENCIES.md? Mac has a recent one (3.9+) as well, I suppose?

### CODE REVIEW

**Comment by mrzzzrm** [comment]
> Except for the coverage, which is why I put 7.1.

:( 

> CLion was still using a vintage version back then.

That must have been a long time ago, I can only remember CLion shipping a *very* recent version. Currently its 3.14.2

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,15 +19,15 @@ function(compiler_not_supported message)
...
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8.2)
         compiler_not_supported("Your GCC version ${CMAKE_CXX_COMPILER_VERSION} is too old.")
     endif()
     if (APPLE)
         # https://software.intel.com/en-us/forums/intel-threading-building-blocks/topic/749200
         compiler_not_supported("We had to drop support for GCC on OS X because it caused segfaults when used with tbb. You can continue, but don't hold us responsible for any segmentation faults.")
     endif()
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0)
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.1)
```
Comment (mrks) [inline CMakeLists.txt]: If you want, we can use 7.0 here and check for 7.1 in coverage.sh

**Comment by mrzzzrm** [comment]
Which ubuntu version does the CI use? Because apparently it doesn't have 7.1, but 7.0.1

image

**Comment by mrks** [comment]
Argh. That's what I meant, sorry. I believe that one is not available in 18.04 though.

**Comment by mrzzzrm** [comment]
I am confused, clang 7.0.1 seems to have generated a working coverage report for 
cf2946d: https://hyrise-ci.epic-hpi.de/job/hyrise/job/hyrise/job/PR-1660/2/execution/node/117/log/?consoleFull

How can I check for 7.1 in coverage.sh if the CI server doesn't - see the screenshot - have 7.1? The CI server runs coverage.sh, right?

**Comment by mrks** [comment]
7.0.1 is good, 7.0 is bad: https://github.com/hyrise/hyrise/pull/1433#issuecomment-458082341
Thus, checking for >= 7.0 when compiling at >= 7.0.1 in coverage.sh should work.

**Comment by mrzzzrm** [comment]
Alright, any pointers on how I check for minimum-version-fulfilled in bloody bash? ;)

**Comment by mrks** [comment]
Nothing native, but there are a bunch of semver functions out there.

**Comment by mrks** [comment]
Probably that doesn’t cover everything, but it’s good enough for now.

**Pull request merged**